test/src/unit-capi-string_dims.cc

```diff
@@ -2133,8 +2133,12 @@ TEST_CASE_METHOD(
   CHECK(r_d_val == "aaccccdddd");
   std::vector<uint64_t> c_d_off = {0, 2, 4, 6};
   CHECK(r_d_off == c_d_off);
-  std::vector<int32_t> c_a = {1, 2, 3, 4};
-  CHECK(r_a == c_a);
+  // The ordering of 'a' is undefined for duplicate dimension
+  // elements. Check both for dimension element "c".
+  std::vector<int32_t> c_a_1 = {1, 3, 2, 4};
```

Comment:
Are these two permutations enough? Do we need instead a generic "vector_contains" which loops over each element?

Reply:
Good question, but the scenario here is a little more nuanced than checking if the results match any permutation. For example, in this scenario: only `{1, 3, 2, 4}` and `{1, 2, 3, 4}` are valid because indexes `1` and `2` correspond to "c" and therefore their values are interchangeable. The values at index `0` and `3` must remain fixed.